### ISSUE TITLE
Loss of precision when plotting large floats in Makie

### ISSUE THREAD

**Issue by malthesr** [MakieOrg/Makie.jl#1196]
I posted an [issue](https://discourse.julialang.org/t/loss-of-precision-when-plotting-large-floats-in-makie/65802) to the Julia discourse, and the helpful replies there seem to suggest that this may be a bug in (or downstream from?) Makie, so I figured I'd file an issue here also since I couldn't find a duplicate in the issue tracker.

For completeness, I'll repeat my issue here: I'm doing something which I've reduced to:

```julia
using GLMakie

fig = Figure()
ax = Axis(fig[1, 1])

x = (0.1:0.1:10) .+ 1e7
y = 1:100

scatter!(ax, x, y)

save("example.png", fig)
``` 

This produces:

![image](https://user-images.githubusercontent.com/25851939/128203704-799262f2-d4f0-4116-9b8f-d1117e035e39.png)

Whereas I'd expected a straight line. As [`johnmyleswhite`](https://discourse.julialang.org/t/loss-of-precision-when-plotting-large-floats-in-makie/65802/2) points out in the discourse, the loss of precision is not in the data itself, but seems to happen somewhere along the plotting machinery.

I'm running Makie 0.15.0 with Julia 0.16.2 on Manjaro Linux.

Thank you for the library, I've found it very enjoyable to use.

**Comment by klaff** [comment]
I just ran into this when plotting oscilloscope data. 5s offset in time value with 100e-9 resolution. `eps(5f0)` gives 477e-9, and accordingly I'm getting 4 or 5 data points mashed together at common x-axis locations.

**Comment by ffreyer** [comment]
See also #780, #931

**Comment by Enlil50** [comment]
Hi, I just ran in the same issue.

Is it possible to solve it for GLMakie too?
I want to use the interactive GUI.

Related: https://discourse.julialang.org/t/glmakie-vs-plots-not-enogh-precision-for-axis-limits/111249

**Comment by asinghvi17** [comment]
Looks fixed on 0.21
![iTerm2 ilN586](https://github.com/MakieOrg/Makie.jl/assets/32143268/4dfb5c28-bf01-4dad-a3e0-59670459811c)
